src/h3/mod.rs

```diff
@@ -351,6 +351,10 @@ impl Connection {
         self.streams
             .insert(stream_id, stream::Stream::new(stream_id, true));
 
+        if conn.grease {
```

Comment:
Why not just do this from inside `send_headers()` so you don't duplicate the code?

Reply:
good observation, done